health-services/PassiveGoals/app/src/main/java/com/example/passivegoals/HealthServicesManager.kt

```diff
@@ -64,26 +58,43 @@ class HealthServicesManager @Inject constructor(
     private val passiveMonitoringClient = healthServicesClient.passiveMonitoringClient
 
     suspend fun hasFloorsAndDailyStepsCapability(): Boolean {
-        val capabilities = passiveMonitoringClient.capabilities.await()
-        return capabilities.supportedDataTypesEvents.containsAll(
+        val capabilities = passiveMonitoringClient.getCapabilitiesAsync().await()
+        return capabilities.supportedDataTypesPassiveGoals.containsAll(
             setOf(
-                DataType.TOTAL_CALORIES,
-                DataType.FLOORS
+                DataType.STEPS_DAILY,
+                DataType.FLOORS_DAILY,
             )
         )
     }
 
-    suspend fun subscribeForGoals() {
+    @OptIn(ExperimentalCoroutinesApi::class)
+    suspend fun subscribeForGoals(): Boolean {
         Log.i(TAG, "Subscribing for goals")
-        val componentName = ComponentName(context, PassiveGoalsReceiver::class.java)
-        // Each goal is a separate subscription.
-        passiveMonitoringClient.registerPassiveGoalCallback(dailyStepsGoal, componentName).await()
-        passiveMonitoringClient.registerPassiveGoalCallback(floorsGoal, componentName).await()
+        val passiveListenerConfig =
+            PassiveListenerConfig.builder().setDailyGoals(setOf(dailyStepsGoal, floorsGoal))
```

Comment:
There was a comment about needing to subscribe to data types for just data as well as goals, for goals to work, but not clear to me if this was resolved as being necessary or not? https://docs.google.com/document/d/1JyHjzxByoccfyn2tGg5GQOlIEymT5YFGeS2lLWC69j4/edit?disco=AAAAgAKiDJk

Do you know?

Reply:
Following up internally. Seems to work without explicitly specifying the extra data types though.